client/js/src/utils.ts

```diff
@@ -239,3 +239,62 @@ export const hardware_types = [
 	"a10g-large",
 	"a100-large"
 ] as const;
+
+export function apply_diff(
+	obj: any,
+	diff: [string, (number | string)[], any][]
+): any {
+	const apply_edit = (
+		target: any,
+		path: (number | string)[],
+		action: string,
+		value: any
+	): any => {
```

Comment:
Is there any reason for this `apply_edit` function to be inside of the `apply_diff` function? It doesn't look like it depends on anything other than its arguments.

Also could we switch this to be a `function apply_edit() { ... }` for consistency?